<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>898</width>
    <height>514</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Serial Port Demo</string>
  </property>
  <property name="windowIcon">
   <iconset resource="res.qrc">
    <normaloff>:/res/res/com_plug01.ico</normaloff>:/res/res/com_plug01.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_13">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>SERIAL PORT SETTINGS</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>SERIAL DEVICE NAME</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLineEdit" name="linePortName">
             <property name="text">
              <string>/dev/ttyUSB0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="topMargin">
              <number>3</number>
             </property>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>DATA BITS</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinDataBits">
               <property name="minimum">
                <number>5</number>
               </property>
               <property name="maximum">
                <number>8</number>
               </property>
               <property name="value">
                <number>8</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <property name="topMargin">
              <number>3</number>
             </property>
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>PARITY</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboParity">
               <property name="currentIndex">
                <number>0</number>
               </property>
               <item>
                <property name="text">
                 <string>None</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Odd</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Even</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <property name="topMargin">
              <number>3</number>
             </property>
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>STOP BITS</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinStopBits">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>2</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <property name="topMargin">
            <number>3</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>BAUDRATE</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QComboBox" name="comboBaudrate">
               <property name="currentIndex">
                <number>3</number>
               </property>
               <item>
                <property name="text">
                 <string>1200</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>2400</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>4800</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>9600</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>19200</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>38400</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>57600</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>115200</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="pbOpenPort">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>OPEN</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbClosePort">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>CLOSE</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
        <zorder></zorder>
        <zorder></zorder>
        <zorder></zorder>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>TX</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QSpinBox" name="spinWriteByte">
               <property name="value">
                <number>51</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pbWriteByte">
               <property name="text">
                <string>WRITE
BYTE</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <item>
              <widget class="QLineEdit" name="lineWriteLine">
               <property name="text">
                <string>HELLO KITTY</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pbWriteLine">
               <property name="text">
                <string>WRITE
LINE</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <spacer name="horizontalSpacer_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="pbSelectFile">
             <property name="text">
              <string>SELECT
FILE</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pbSendFile">
             <property name="text">
              <string>SEND
FILE</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLineEdit" name="lineFileName">
           <property name="minimumSize">
            <size>
             <width>300</width>
             <height>0</height>
            </size>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>RX</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QTextBrowser" name="textBrowserRX"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QRadioButton" name="rbTxt">
           <property name="text">
            <string>TXT</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbHex">
           <property name="text">
            <string>HEX</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rbDec">
           <property name="text">
            <string>DEC</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pbClearRx">
           <property name="text">
            <string>CLEAR</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
